### PR TITLE
CBL-4462: sync function missing from sg_setup script

### PR DESCRIPTION
**Pull request by callumbirks**
Add sync functions to sg_setup script.
Remove assertions from SGTestUser because some tests caused it to fail. 
Change test tags of a test that shouldn't be grouped with the other 'ReplicatorCollectionTest's.

### CODE REVIEW

File: Replicator/tests/SGTestUser.hh [hunk 1]
```diff
@@ -24,6 +24,7 @@
 #include "SG.hh"
 #include "Error.hh"
 #include "HTTPLogic.hh"
+#include "c4Private.h"
```
Comment (jianminzhao) [inline Replicator/tests/SGTestUser.hh:27]: Why is this needed now?

**Review by jianminzhao — APPROVED** [review]

File: Replicator/tests/SGTestUser.hh (hunk 1 above)
Comment (callumbirks) [reply Replicator/tests/SGTestUser.hh:27]: Hm, good point, it's probably not
